lib/kitchen/directions/bake_notes/bake_numbered_notes.rb

```diff
@@ -26,6 +29,28 @@ def self.bake_note(note:)
           HTML
         )
       end
+
+      def self.bake_note_exercise(note:)
+        exercise = note.exercises.first
+        exercise.solution ? exercise.add_class('os-hasSolution unnumbered') : exercise.add_class('unnumbered')
+        # bake problem
+        exercise.problem.wrap_children('div', class: 'os-problem-container')
+        exercise.problem.search('strong').first&.trash
+        return unless exercise.solution
```

Comment:
We call `exercise.solution` here and then 3 more times below.  Let's memoize it.  Until we merge in my "little bit of caching" PR, every time we call `.solution` it does a new search for it.  

So either

```ruby
solution = exercise.solution
return unless solution
```

or

```ruby
return unless (solution = exercise.solution)
```

and then replace `exercise.solution` with `solution`.  Also makes the code a little more compact